include/mbgl/map/map_observer.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace mbgl {
+
+class MapObserver {
+public:
+    static MapObserver& nullObserver() {
+        static MapObserver mapObserver;
+        return mapObserver;
+    }
+
+    virtual void onRegionWillChange() {}
```

Comment:
Ack. For now I'm abstaining from renaming the public APIs for Qt & Android. For iOS & macOS I'm adding entries to `CHANGELOG.md` about the renaming.